lib/services/presets.go

```diff
@@ -202,16 +205,79 @@ func NewPresetAuditorRole() types.Role {
 	return role
 }
 
+// NewPresetReviewerRole returns a new pre-defined role for reviewer. The
+// reviewer will be able to review all access requests.
+func NewPresetReviewerRole() types.Role {
+	if modules.GetModules().BuildType() != modules.BuildEnterprise {
+		return nil
+	}
+
+	role := &types.RoleV6{
+		Kind:    types.KindRole,
+		Version: types.V6,
+		Metadata: types.Metadata{
+			Name:        teleport.PresetReviewerRoleName,
+			Namespace:   apidefaults.Namespace,
+			Description: "Review access requests",
+			Labels: map[string]string{
+				types.TeleportManagedLabel: types.IsManaged,
+			},
+		},
+		Spec: types.RoleSpecV6{
+			Allow: types.RoleConditions{
+				ReviewRequests: defaultAllowAccessReviewConditions(true)[teleport.PresetReviewerRoleName],
+			},
+		},
+	}
+	role.SetLogins(types.Allow, []string{"no-login-" + uuid.New().String()})
+	return role
+}
+
+// NewPresetRequesterRole returns a new pre-defined role for requester. The
+// requester will be able to request all resources.
+func NewPresetRequesterRole() types.Role {
+	if modules.GetModules().BuildType() != modules.BuildEnterprise {
+		return nil
+	}
+
+	role := &types.RoleV6{
+		Kind:    types.KindRole,
+		Version: types.V6,
+		Metadata: types.Metadata{
+			Name:        teleport.PresetRequesterRoleName,
+			Namespace:   apidefaults.Namespace,
+			Description: "Request all resources",
+			Labels: map[string]string{
+				types.TeleportManagedLabel: types.IsManaged,
+			},
+		},
+		Spec: types.RoleSpecV6{
+			Allow: types.RoleConditions{
+				Request: defaultAllowAccessRequestConditions(true)[teleport.PresetRequesterRoleName],
+			},
+		},
+	}
+	role.SetLogins(types.Allow, []string{"no-login-" + uuid.New().String()})
+	return role
+}
+
 // NewPresetGroupAccessRole returns a new pre-defined role for group access -
 // a role used for requesting and reviewing user group access.
```

Comment:
does it allow requesting and reviewing?

Reply:
It doesn't allow it, but it supports it -- it's used by reviewer and requester roles for enabling user group requests and user group reviews.